File: Napakalaki JAVA/src/napakalaki_java/Dice.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki_java;

/**
 *
 * @author jumacasni
 */
public class Dice {
    
    private static Dice instance = null;
    
    private Dice() {
    }
    
    public static Dice getInstance(){
        if(instance == null)
            instance = new Dice();
        
        return instance;
    }
    
    public int nextNumber(){
        int numeroAleatorio = (int) (Math.random()*6+1);
        
        return numeroAleatorio;
    }
}

File: Napakalaki RUBY/lib/dice.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require "singleton"

module NapakalakiGame
  class Dice
  
  include Singleton
    
  def nextNumber()
    num_aleat = rand(1..6)
    return num_aleat
  end

  end
end

